rebasehelper/specfile.py

```diff
@@ -352,31 +352,36 @@ def sanitize(tag):
                     return '{0}0'.format(tokens[0])
                 return '{0}{1}'.format(tokens[0], int(tokens[1]))
             return tag.capitalize()
-        result = {}
-        parsed = self.spc.parsed.split('\n')
+        result: Dict[str, List[Tag]] = {}
+        parsed = self.spc.parsed.split('\n')  # FIXME: we should split this on sections too
```

Comment:
Do I understand correctly that the FIXME talks about handling conditionalized sections correctly? If so, the fix should be fairly simple. Checking whether the section header ( the first element in each element of `SpecContent.sections`) is in `parsed` should work in this case. Perhaps it would be better to create a `set` from `parsed` to achieve better time performance

Reply:
No, it's about finding matching tags in the right section. There could be duplicate tags in different sections and the validity check could be inaccurate otherwise.